examples/Migration/CampaignManagement/CreateCompleteCampaignGoogleAdsApiOnly.php

```diff
@@ -0,0 +1,499 @@
+
+        $response =
+            $googleAdsServiceClient->search($customerId, $query, ['pageSize' => self::PAGE_SIZE]);
+
+        return $response->getIterator()->current()->getCampaignBudget();
+    }
+
+    /**
+     * Creates a campaign.
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param string $customerId the client customer ID without hyphens
+     * @param CampaignBudget $campaignBudget the campaign budget
+     * @return Campaign the newly created campaign
+     */
+    private static function createCampaign(
+        GoogleAdsClient $googleAdsClient,
+        string $customerId,
+        CampaignBudget $campaignBudget
+    ) {
+        $trueValue = new BoolValue(['value' => true]);
+        $falseValue = new BoolValue(['value' => false]);
+
+        $startDate = new StringValue(['value' => date('Ymd', strtotime('+1 day'))]);
+        $endDate = new StringValue(['value' => date('Ymd', strtotime('+1 month'))]);
+
+        // Configures the campaign network options.
+        $networkSettings = new NetworkSettings([
+            'target_google_search' => $trueValue,
+            'target_search_network' => $trueValue,
+            'target_content_network' => $falseValue,
+            'target_partner_search_network' => $falseValue
+        ]);
+
+        $budgetResourceName = $campaignBudget->getResourceName();
+
+        $campaign = new Campaign([
+            'name' => new StringValue(['value' => 'Interplanetary Cruise #' . uniqid()]),
+            'advertising_channel_type' => AdvertisingChannelType::SEARCH,
+            // Recommendation: Set the campaign to PAUSED when creating it to prevent
+            // the ads from immediately serving. Set to ENABLED once you've added
+            // targeting and the ads are ready to serve.
+            'status' => CampaignStatus::PAUSED,
+            // Sets the bidding strategy and budget.
+            'manual_cpc' => new ManualCpc(),
+            'campaign_budget' => $budgetResourceName,
+            // Adds the network settings configured above.
+            'network_settings' => $networkSettings,
+            // Optional: Sets the start and end dates.
+            'start_date' => $startDate,
+            'end_date' => $endDate
+        ]);
+
+        // Creates a campaign operation.
+        $campaignOperation = new CampaignOperation();
+        $campaignOperation->setCreate($campaign);
+
+        // Issues a mutate request to add campaigns.
+        $campaignServiceClient = $googleAdsClient->getCampaignServiceClient();
+        $campaignResponse = $campaignServiceClient->mutateCampaigns(
+            $customerId,
+            [$campaignOperation]
+        );
+
+        $campaignResourceName = $campaignResponse->getResults()[0]->getResourceName();
+        $newCampaign = self::getCampaign($googleAdsClient, $customerId, $campaignResourceName);
+
+        printf("Added campaign named '%s'%s", $newCampaign->getName()->getValue(), PHP_EOL);
```

Comment:
Added campaign named '%s'.%s

<-- Need to have a period.

Reply:
Done.